sos/sosreport.py

```diff
@@ -845,9 +845,9 @@ def _setup_logging(self):
         # main soslog
         self.soslog = logging.getLogger('sos')
         self.soslog.setLevel(logging.DEBUG)
-        self.sos_log_file = self.get_temp_file()
-        self.sos_log_file.close()
-        flog = logging.FileHandler(self.sos_log_file.name)
+        fd, self.sos_log_file = self.get_temp_file()
+        fd.close()
+        flog = logging.FileHandler(self.sos_log_file)
```

Comment:
This looks completely WTF... We are asking our tempfiles class for a new, temporary FileObject, which it returns, then we close it, and pass it to another function that re-opens it based on the path? That's completely insane. It also re-introduces the TOCTOU race that the original change was intended to defeat (since there is now a race window between `fd.close()` and the  re-open taking place inside `flog = logging.FileHandler(self.sos_log_file)`.